pkg/autoscaling/calculation.go

```diff
@@ -307,35 +326,55 @@ func getScaledTiDBGroups(etcdClient *clientv3.Client, healthyInstances []instanc
 				zap.String("tidb-address", instance.address))
 			return nil
 		}
-		v := tidb.getLabelValue(groupLabelKey)
-		buildPlanMap(planMap, v, instance.address)
+
+		groupName := tidb.getLabelValue(groupLabelKey)
+		if !isAutoScaledGroup(groupName) {
+			continue
+		}
+
+		buildPlanMap(planMap, groupName, instance.address)
+		resourceType := tidb.getLabelValue(resourceTypeLabelKey)
+		if _, ok := resourceTypeMap[groupName]; !ok {
+			if resourceType == "" {
+				log.Warn("tidb is in auto-scaled group but has no resource type label, exit auto-scaling calculation", zap.String("tidb-address", instance.address))
+				return nil
+			}
+			resourceTypeMap[groupName] = resourceType
+		}
 	}
-	return buildPlans(planMap, TiDB)
+	return buildPlans(planMap, resourceTypeMap, TiDB)
+}
+
+func isAutoScaledGroup(groupName string) bool {
+	return len(groupName) > len(autoScalingGroupLabelKeyPrefix) && strings.HasPrefix(groupName, autoScalingGroupLabelKeyPrefix)
 }
 
 func buildPlanMap(planMap map[string]map[string]struct{}, groupName, address string) {
-	if len(groupName) > len(autoScalingGroupLabelKeyPrefix) &&
-		strings.HasPrefix(groupName, autoScalingGroupLabelKeyPrefix) {
-		if component, ok := planMap[groupName]; ok {
-			component[address] = struct{}{}
-		} else {
-			planMap[groupName] = map[string]struct{}{
-				address: {},
-			}
+	if component, ok := planMap[groupName]; ok {
+		component[address] = struct{}{}
+	} else {
+		planMap[groupName] = map[string]struct{}{
+			address: {},
 		}
 	}
 }
 
-func buildPlans(planMap map[string]map[string]struct{}, componentType ComponentType) []*Plan {
+func buildPlans(planMap map[string]map[string]struct{}, resourceTypeMap map[string]string, componentType ComponentType) []*Plan {
 	var plans []*Plan
-	for groupLabel, groupInstances := range planMap {
+	for groupName, groupInstances := range planMap {
+		resourceType := resourceTypeMap[groupName]
 		plans = append(plans, &Plan{
-			Component: componentType.String(),
-			Count:     uint64(len(groupInstances)),
+			Component:    componentType.String(),
+			Count:        uint64(len(groupInstances)),
+			ResourceType: resourceType,
```

Comment:
Do we still need it since we have already recorded the resource type in the label?

Reply:
The API definition requires this field

Reply:
Yes, but if it's duplicated, we can change the API.